src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -424,22 +424,35 @@ free_unmap_bo(xclDeviceHandle handle, xclBufferHandle boh, void * boptr, size_t
  * helper function for P2P test
  */
 static bool
-p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
+p2ptest_set_or_cmp(char *boptr, size_t size, char *valid_data, size_t valid_data_size, bool set)
 {
   int stride = xrt_core::getpagesize();
 
   assert((size % stride) == 0);
+  assert(size > valid_data_size);
```

Comment:
Using the vector, this code becomes:
```suggestion
  assert(size > validData.size());
```